### PR TITLE
[NGPDCP-2823] Add new variant for badge component

### PR DESCRIPTION
**Pull request by hoapn-fsoft**
# Add new variant for badge component

## Description

**1) Screenshot

image

**2) Changes**

**3) Impact**

## Check list

- [x] 1. Did you start and verify your changes?
- [x] 2. Did you run build to make sure that your changes can be built successfully?
- [x] 3. No !important flag, inline style in css
- [x] 4. No boilerplate codes (1)
- [x] 5. No duplicate code that exist in common functions?
- [x] 6. All of defined variables should have default value, check null and undefined?
- [x] 7. Use meaningful name for variables, no suffix 1,2,... Describes reference information for easier to understand what's inside. (2)
- [x] 9. Unsubscribed all of subscribers and even listeners when you don't need them.

```
(1)
Boilerplate codes is duplicated multiple times a bunch of the same code. This should be a common function to reuse through your code or you can use generic class / methods or design pattern to avoid the Boilerplate codes.
```

```
(2)
Follow theo coding convention
NO acronym variable name:
WRONG: el, elm
RIGHT: element
Your code can be read as a sentence
```

### CODE REVIEW

File: packages/react-compass/src/badge/badge.styles.ts [hunk 1]
```diff
@@ -49,6 +49,39 @@ export const StyledBadge = styled('div', {
       primary: {},
       secondary: {},
       outline: {},
+      h5: {
+        background: '$white',
```
Comment (pieliedie) [inline packages/react-compass/src/badge/badge.styles.ts:53]: image
dark mode

File: packages/react-compass/src/badge/badge.tsx [hunk 2]
```diff
@@ -7,6 +8,8 @@ import {BadgeVariantProps, StyledBadge} from './badge.styles'
 interface Props extends StyledComponentProps {
   label?: string
   icon?: boolean | React.ReactNode
+  destination?: string
+  revert?: boolean
```
Comment (pieliedie) [inline packages/react-compass/src/badge/badge.tsx:12]: `isReverted`

**Review by pieliedie — CHANGES_REQUESTED** [review]

File: packages/react-compass/src/badge/badge.tsx (hunk 2 above)
Comment (hoapn-fsoft) [reply packages/react-compass/src/badge/badge.tsx]: Updated, thank @pieliedie

File: packages/react-compass/src/badge/badge.styles.ts (hunk 1 above)
Comment (hoapn-fsoft) [reply packages/react-compass/src/badge/badge.styles.ts]: updated

**Review by pieliedie — APPROVED** [review]

**Review by Tai-Chi-Bui — APPROVED** [review]

**Pull request merged**